autotest/cpp/test_gdal_jpegxr.cpp

```diff
@@ -0,0 +1,389 @@
+        GDALRasterBandH hBand1 = GDALGetRasterBand(ds_.h, 1);
+        CPLErr eErr = GDALGetRasterStatistics(hBand1, false, true, &dfMin, &dfMax, &dfMean, &dfStdD);
+        ensure_equals(eErr, CE_None);
+    }
+
+    // READ: 24bpp BGR
+    template<>
+    template<>
+    void object::test<8>()
+    {
+        std::string file = get_file_path("lenna-256x256-24bpp-BGR.jxr");
+        ds_ = GDALOpen(file.c_str(), GA_ReadOnly);
+        ensure(ds_.h);
+
+        double dfMin(0), dfMax(0), dfMean(0), dfStdD(0);
+        for (int i = 0; i < GDALGetRasterCount(ds_.h); i++)
+        {
+            GDALRasterBandH hBand1 = GDALGetRasterBand(ds_.h, i + 1);
+            CPLErr eErr = GDALGetRasterStatistics(hBand1, false, true, &dfMin, &dfMax, &dfMean, &dfStdD);
+            ensure_equals(eErr, CE_None);
+        }
+    }
+
+    // READ: 24bpp RGB
+    template<>
+    template<>
+    void object::test<9>()
+    {
+        std::string file = get_file_path("mandril-512x512-24bpp-RGB.jxr");
+        ds_ = GDALOpen(file.c_str(), GA_ReadOnly);
+        ensure(ds_.h);
+
+        double dfMin(0), dfMax(0), dfMean(0), dfStdD(0);
+        for (int i = 0; i < GDALGetRasterCount(ds_.h); i++)
+        {
+            GDALRasterBandH hBand1 = GDALGetRasterBand(ds_.h, i + 1);
+            CPLErr eErr = GDALGetRasterStatistics(hBand1, false, true, &dfMin, &dfMax, &dfMean, &dfStdD);
+            ensure_equals(eErr, CE_None);
+        }
+    }
+
+    // READ: 32bpp BGRA
+    template<>
+    template<>
+    void object::test<10>()
+    {
+        std::string file = get_file_path("lenna-256x256-32bpp-BGRA.jxr");
+        ds_ = GDALOpen(file.c_str(), GA_ReadOnly);
+        ensure(ds_.h);
+
+        double dfMin(0), dfMax(0), dfMean(0), dfStdD(0);
+        for (int i = 0; i < GDALGetRasterCount(ds_.h); i++)
+        {
+            GDALRasterBandH hBand1 = GDALGetRasterBand(ds_.h, i + 1);
+            CPLErr eErr = GDALGetRasterStatistics(hBand1, false, true, &dfMin, &dfMax, &dfMean, &dfStdD);
+            ensure_equals(eErr, CE_None);
+        }
+    }
+
+    // READ: 32bpp RGBA
+    template<>
+    template<>
+    void object::test<11>()
+    {
+        std::string file = get_file_path("lenna-256x256-32bpp-RGBA.jxr");
+        ds_ = GDALOpen(file.c_str(), GA_ReadOnly);
+        ensure(ds_.h);
+
+        double dfMin(0), dfMax(0), dfMean(0), dfStdD(0);
+        for (int i = 0; i < GDALGetRasterCount(ds_.h); i++)
+        {
+            GDALRasterBandH hBand1 = GDALGetRasterBand(ds_.h, i + 1);
+            CPLErr eErr = GDALGetRasterStatistics(hBand1, false, true, &dfMin, &dfMax, &dfMean, &dfStdD);
+            ensure_equals(eErr, CE_None);
+        }
+    }
+
+    // COPY ////////////////////////////////////////////////////////////////////
+
+    // COPY: 8bpp Gray, no options/encoding defaults
+    template<>
+    template<>
+    void object::test<12>()
+    {
+            std::string src_file = get_file_path("lenna-256x256-8bpp-Gray.tif");
```

Comment:
@schwehr Tests are not a place where one should strive for saving code or functional cleverness. Tests must be simple, explicit and self-descriptive. String concatenation is an unnecessary complexity, might even increase obscurity of a test function. KIS.